File: DSA/Ayush_Mishra_590014844/Day_38_Question_2.py
class MinStack:
    def __init__(self):
        self.stack = []
        self.minStack = []
    
    def push(self, val):
        self.stack.append(val)
        if not self.minStack or val <= self.minStack[-1]:
            self.minStack.append(val)
    
    def pop(self):
        if self.stack:
            if self.stack[-1] == self.minStack[-1]:
                self.minStack.pop()
            self.stack.pop()
    
    def getMin(self):
        return self.minStack[-1] if self.minStack else None

s = MinStack()
while True:
    print("\n1. Push\n2. Pop\n3. Get Min\n4. Exit")
    ch = int(input("Enter choice: "))
    if ch == 1:
        val = int(input("Enter value: "))
        s.push(val)
    elif ch == 2:
        s.pop()
    elif ch == 3:
        print("Minimum element:", s.getMin())
    elif ch == 4:
        break
    else:
        print("Invalid choice")

File: DSA/Ayush_Mishra_590014844/Day_38_Question_2.c
#include <stdio.h>
#define MAX 100

int main() {
    int stack[MAX], minStack[MAX];
    int top = -1, minTop = -1, ch, x;

    while(1) {
        printf("\n1. Push\n2. Pop\n3. Get Min\n4. Exit\nEnter your choice: ");
        scanf("%d", &ch);

        if(ch == 1) {
            printf("Enter element to push: ");
            scanf("%d", &x);
            stack[++top] = x;
            if(minTop == -1 || x <= minStack[minTop]) {
                minStack[++minTop] = x;
            }
        } else if(ch == 2) {
            if(top == -1) {
                printf("Stack is empty\n");
            } else {
                if(stack[top] == minStack[minTop]) {
                    minTop--;
                }
                top--;
            }
        } else if(ch == 3) {
            if(minTop == -1) {
                printf("Stack is empty\n");
            } else {
                printf("Minimum element: %d\n", minStack[minTop]);
            }
        } else if(ch == 4) {
            break;
        } else {
            printf("Invalid choice\n");
        }
    }

    return 0;
}
